openshift/backup/backup-build.yaml

```diff
@@ -0,0 +1,74 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: ${NAME}-build-template
+  creationTimestamp: null
+objects:
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: ${NAME}
+  - kind: BuildConfig
+    apiVersion: v1
+    metadata:
+      name: ${NAME}
+      labels:
+        app: ${NAME}
+    spec:
+      triggers:
+        - type: ImageChange
+        - type: ConfigChange
+      runPolicy: Serial
+      source:
+        type: Git
+        git:
+          uri: ${GIT_REPO_URL}
+          ref: ${GIT_REF}
+        contextDir: ${SOURCE_CONTEXT_DIR}
+      strategy:
+        type: Docker
+        dockerStrategy:
+          from:
+            kind: DockerImage
+            name: ${BASE_IMAGE_FOR_BUILD}
+          dockerfilePath: ${DOCKER_FILE_PATH}
+      output:
+        to:
+          kind: ImageStreamTag
+          name: ${NAME}:${OUTPUT_IMAGE_TAG}
```

Comment:
do we need to create an ImageStreamTag as an object explicitly (like ImageStream), or maybe it gets created if the stream is created?

Reply:
It's part of the documentation provided by https://github.com/BCDevOps/backup-container

I'm just taking the scripts from the repository that I've used to create the container (hopefully) and document the process in our repo.

Generally when you create an image stream and push an image to it, the imagestreamtag is automatically generated. If I trace my steps backwards in the tools namespace, there is a buildconfig that generates an image that is output into this imagestream, therefore the tag is created on push.